app/src/main/java/com/duckduckgo/app/browser/useragent/UserAgentProvider.kt

```diff
@@ -96,10 +102,20 @@ class UserAgentProvider constructor(@Named("defaultUserAgent") private val defau
     }
 
     private fun getSafariComponentFromUserAgent(): String? {
-        val matches = AgentRegex.safari.find(defaultUserAgent.get()) ?: return null
+        val matches = AgentRegex.safari.find(defaultUserAgent()) ?: return null
         return matches.groupValues.last()
     }
 
+    private fun defaultUserAgent(): String = runCatching {
+        defaultUserAgent.get()
+    }.let { result ->
+        if (result.isFailure) {
+            pixel.fire(AppPixelName.WEBVIEW_DEFAULT_USER_AGENT_CRASH)
+            return fallbackDefaultUA
+        }
+        return@let result.getOrDefault(fallbackDefaultUA)
+    }
+
```

Comment:
I think this should rather be in the `BrowserModule`. There is no need to make changes to this calss.

```kotlin
    @Provides
    @SingleInstanceIn(AppScope::class)
    fun userAgentProvider(@Named("defaultUserAgent") defaultUserAgent: Provider<String>, deviceInfo: DeviceInfo): UserAgentProvider {
        try {
            return UserAgentProvider(defaultUserAgent, deviceInfo)
        } catch (e: Exception) {
            return ""
        }
    }
```

Reply:
Did you mean change the `DefaultUserAgentModule`? :P

```
@ContributesTo(AppScope::class)
@Module
class DefaultUserAgentModule {
    @SingleInstanceIn(AppScope::class)
    @Provides
    @Named("defaultUserAgent")
    fun provideDefaultUserAgent(context: Context): String {
       try {
           return WebSettings.getDefaultUserAgent(context)
        } {catch (e: Exception) {
            return ""
        }
    }
}

Reply:
Yup 👍  to this

Reply:
I'm ok not changing the class, however, I don't think is a good idea to provide an empty string as a fallback.

WDTY about wrapping `WebSettings.getDefaultUserAgent(context)` into in our own class that will provide a default UA?

Reply:
I think the empty string was an example, we should return the actual UA there.

Reply:
I've discussed this with @aitorvs. If we don't need to send a pixel, we can follow this approach.